plugins/node/opentelemetry-instrumentation-hapi/src/instrumentation.ts

```diff
@@ -386,10 +396,19 @@ export class HapiInstrumentation extends InstrumentationBase {
         const span = instrumentation.tracer.startSpan(metadata.name, {
           attributes: metadata.attributes,
         });
-        const res = await oldHandler(request, h, err);
-        span.end();
-
-        return res;
+        try {
+          const res = await oldHandler(request, h, err);
+          return res;
```

Comment:
Don't need the assignment nor `await` here. Just `return oldHandler(...);`

Reply:
removed the unneeded assignment.
as for the await, it is needed as catch won't work on promises that are not awaited, and I also need to wait to close the span after the action was completed.
maybe i should add a test that includes an async handler?
Also, there is a  handy eslint rule that covers that https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/return-await.md.